src/groups/groups.service.spec.ts

```diff
@@ -46,6 +46,32 @@ describe('GroupsService', () => {
     expect(service).toBeDefined();
   });
 
+  describe('createGroup', () => {
+    const groupName = 'testGroup';
+    const displayname = 'Group Test';
+    beforeEach(() => {
+      jest
+        .spyOn(groupRepo, 'save')
+        .mockImplementationOnce(async (group: Group): Promise<Group> => group);
+    });
+    it('works', async () => {
+      const user = await service.createGroup(groupName, displayname);
+      expect(user.name).toEqual(groupName);
+      expect(user.displayName).toEqual(displayname);
+    });
+    it('fails if username is already taken', async () => {
+      jest.spyOn(groupRepo, 'save').mockImplementationOnce(() => {
+        throw new Error();
```

Comment:
This looks a bit suspicious. With this mock, `groupRepo.save` never succeeds, which should mean that `groupService.createGroup` also never succeeds.